tests/unit/spec/base/mozilla-client.js

```diff
@@ -9,6 +9,172 @@ describe('mozilla-client.js', function() {
+        });
+
+        it('should not consider Iceweasel to be Firefox', function() {
+            expect(window.Mozilla.Client._isFirefox('Mozilla/5.0 (X11; Linux x86_64; rv:17.0) Gecko/20121202 Firefox/17.0 Iceweasel/17.0.1')).toBeFalsy();
+        });
+
+    });
+
+    describe('_isFirefoxDesktop', function () {
+
+        it('should return true for Firefox on Desktop', function() {
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (Windows NT x.y; rv:10.0) Gecko/20100101 Firefox/10.0')).toBeTruthy();
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (Macintosh; Intel Mac OS X x.y; rv:10.0) Gecko/20100101 Firefox/10.0')).toBeTruthy();
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (X11; Linux i686; rv:10.0) Gecko/20100101 Firefox/10.0')).toBeTruthy();
+        });
+
+        it('should return false for Firefox on Android', function() {
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (Android; Mobile; rv:26.0) Gecko/26.0 Firefox/26.0')).toBeFalsy();
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (Android; Tablet; rv:26.0) Gecko/26.0 Firefox/26.0')).toBeFalsy();
+        });
+
+        it('should return false for Firefox on iPhone', function() {
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (iPhone; CPU iPhone OS 8_3 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) FxiOS/1.1 Mobile/12F69 Safari/600.1.4')).toBeFalsy();
+        });
+
+        it('should return false for Firefox OS', function() {
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (Mobile; rv:26.0) Gecko/26.0 Firefox/26.0')).toBeFalsy();
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (Tablet; rv:26.0) Gecko/26.0 Firefox/26.0')).toBeFalsy();
+        });
+
+        it('should return false for other Gecko-based browsers', function() {
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (Macintosh; U; PPC Mac OS X 10.4; en; rv:1.9.2.24) Gecko/20111114 Camino/2.1 (like Firefox/3.6.24)')).toBeFalsy();
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (Macintosh; Intel Mac OS X 10.9; rv:25.0) Gecko/20100101 Firefox/25.0 SeaMonkey/2.22.1')).toBeFalsy();
+        });
+
+        it('should return false for other browsers', function() {
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1; Trident/4.0; .NET CLR 2.0.50727; .NET CLR 3.0.04506.648; .NET CLR 3.5.21022; .NET CLR 3.0.4506.2152; .NET CLR 3.5.30729)')).toBeFalsy();
+            expect(window.Mozilla.Client._isFirefoxDesktop('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/29.0.1547.57 Safari/537.36')).toBeFalsy();
+        });
+
+    });
+
+    describe('_isFirefoxAndroid', function () {
```

Comment:
It would be good to add a test for the Firefox iOS user agent here to make sure it returns false.